### PR TITLE
Fixed bounds check on large views.

### PR DESCRIPTION
**Pull request by MoFtZ**
Fixes https://github.com/m4rs-mt/ILGPU/issues/429.

When injection assertions into Cuda kernels, the `index` should be converted to `long`. Otherwise, if the length of the `ArrayView` is larger than `int.MaxValue`, the upper 32 bits will be truncated.

QUESTION: What should we do about the differences between CPU and Cuda accelerators?
On the CPU, the assertion will check if `ArrayView.Length` is less than `int.MaxValue`. This is not checked on the GPU.

### CODE REVIEW

**Comment by m4rs-mt** [comment]
@MoFtZ I don't think we can generally do anything about these differences when using the .Net debuggers; since we want to run the original code. At this stage of the compiler, we can't use the IL front end properly because we are already generating code...

**Review by m4rs-mt — APPROVED** [review]

**Comment by m4rs-mt** [comment]
@MoFtZ Should I merge this PR or wait until we have discussed how to proceed regarding the different assertion checks?

**Comment by MoFtZ** [comment]
I'm happy to wait. There is still time before the next release of ILGPU.

I'm wondering if we should make the CPU assertions match the GPU assertions? Rather than using `AssertIntIndexRange`, which causes the CPU code to assert more often, we check that the `index` is within the length of the ArrayView.

**Comment by m4rs-mt** [comment]
I think you are absolutely right and we should reconcile these assertions.

File: Src/ILGPU/ArrayView.cs [hunk 1]
```diff
@@ -229,11 +229,7 @@ public ArrayView(ArrayViewSource source, long index, long length)
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             [ViewIntrinsic(ViewIntrinsicKind.GetViewElementAddressByIndex)]
-            get
-            {
-                IndexTypeExtensions.AssertIntIndex(Length);
```
Comment (m4rs-mt) [inline Src/ILGPU/ArrayView.cs:234]: QUESTION: Don't we have to adapt these assertions to match the ones you have added to the code generator?

Comment (MoFtZ) [reply Src/ILGPU/ArrayView.cs:234]: The assertion already exists in the ArrayView's indexer that accepts a `long` parameter.

https://github.com/m4rs-mt/ILGPU/blob/20342179ec67ffddce8bf9d421339cc72f4bbac2/Src/ILGPU/ArrayView.cs#L264-L277

**Review by m4rs-mt — APPROVED** [review]

**Pull request merged**